src/modules/scraper/service.ts

```diff
@@ -28,138 +28,97 @@ export class ScraperService {
   }
 
   public async run() {
+    this.publishBlocks(ChainIds.mainnet, 10);
+    this.publishBlocks(ChainIds.arbitrum, 4);
+    this.publishBlocks(ChainIds.optimism, 4);
+    this.publishBlocks(ChainIds.polygon, 6);
+    this.publishMerkleDistributorBlocks(30);
+  }
+
+  public async publishBlocks(chainId: number, interval: number) {
     while (true) {
       try {
-        const latestBlocks = await this.getLatestBlocks();
-
-        if (this.appConfig.values.enableSpokePoolsEventsProcessing) {
-          await this.publishBlocks(latestBlocks);
-        }
-        if (this.appConfig.values.enableMerkleDistributorEventsProcessing) {
-          await this.publishMerkleDistributorBlocks(latestBlocks);
+        const blockNumber = await this.providers.getProvider(chainId).getBlockNumber();
+        this.logger.log(`latest block chainId: ${chainId} ${blockNumber}`);
+        const range = await this.determineBlockRange(
+          chainId,
+          blockNumber,
+          this.appConfig.values.web3.spokePoolContracts[chainId].startBlockNumber,
+          this.processedBlockRepository,
+        );
+        if (range) {
```

Comment:
A small comment either here or on determineBlockRange to explain when a caller would expect null/undefined would be helpful. My understanding from the code is that it only happens when from > to, right, which would mean we haven't moved forward since the last run?

Reply:
yes. Basically it is either returning undefined or throwing an error. Anyway I added a comment for now